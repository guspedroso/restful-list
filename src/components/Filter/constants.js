export const constants = {
    SET: `FILTER_SET`
};
