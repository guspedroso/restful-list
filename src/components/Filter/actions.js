import { constants } from './constants';

const set = (filterValue) => dispatch => {
    dispatch({ type: constants.SET, filterValue });
};

export const actions = {
    set
};
