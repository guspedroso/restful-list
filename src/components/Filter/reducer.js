import { constants } from './constants';

export const filter = (state={}, action) => {
    switch (action.type) {
        case constants.SET:
            return {
                filterValue: action.filterValue
            }
        default:
            return state
    }
};
