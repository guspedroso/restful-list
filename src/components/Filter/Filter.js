import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Form } from 'react-bootstrap';
import { actions } from './actions';

class Filter extends Component {
    handleChange = (event) => {
        const { value } = event.target;
        const { set } = this.props;

        set(value);
    }

    render() {
        const { filterValue, placeholder } = this.props;

        return (
            <Form.Group>
                <Form.Control 
                    type='text'
                    placeholder={placeholder}
                    name={filterValue}
                    onChange={this.handleChange}
                />
            </Form.Group>
        );
    }
}

Filter.propTypes = {
    filterValue: PropTypes.string.isRequired,
    placeholder: PropTypes.string.isRequired
};

Filter.defaultProps = {
    filterValue: '',
    placeholder: 'Filter...'
};

const mapStateToProps = (state, ownProps) => {
    const { filter } = state;
    const { filterValue } = filter;

    return {
        filterValue
    }
};

const mapDispatchToProps = dispatch => {
    return {
        set: (filterValue) => dispatch(actions.set(filterValue))
    }
};

export default connect(mapStateToProps, mapDispatchToProps)(Filter);
