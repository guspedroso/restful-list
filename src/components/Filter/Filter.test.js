import React from 'react';
import ReactDOM from 'react-dom';
import Filter from './Filter';
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';

it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
        <Filter setAction={() => {}} />, div);
    ReactDOM.unmountComponentAtNode(div);
});

test('Snapshot works', () => {
    let component = renderer.create(
        <Filter setAction={() => {}} />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
