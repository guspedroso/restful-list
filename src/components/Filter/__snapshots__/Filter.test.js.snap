// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot works 1`] = `
<div
  className="form-group"
>
  <input
    className="form-control"
    name=""
    onChange={[Function]}
    placeholder="Filter..."
    type="text"
  />
</div>
`;
